towerlib/entities/inventory_script.py

```diff
@@ -0,0 +1,152 @@
+    """Models the inventory script entity of ansible tower."""
+
+    def __init__(self, tower_instance, data):
+        Entity.__init__(self, tower_instance, data)
+
+    @property
+    def name(self):
+        """The name of the inventory script.
+
+        Returns:
+            string: The name of the inventory script.
+
+        """
+        return self._data.get('name')
+
+    @name.setter
+    def name(self, value):
+        """Update the name of the inventory script.
+
+        Returns:
+            None:
+
+        """
+        max_characters = 512
+        conditions = [validate_max_length(value, max_characters)]
+        if all(conditions):
+            self._update_values('name', value)
+        else:
+            raise InvalidValue(('{value} is invalid. Condition max_characters must be '
+                                'less or equal to {max_characters}').format(value=value,
+                                                                            max_characters=max_characters))
+
+    @property
+    def description(self):
+        """The description of the inventory script.
+
+        Returns:
+            string: The description of the inventory script.
+
+        """
+        return self._data.get('description')
+
+    @description.setter
+    def description(self, value):
+        """Update the first name of the inventory script.
+
+        Returns:
+            None:
+
+        """
+        self._update_values('description', value)
+
+    @property
+    def script(self):
+        """The script of the inventory script.
+
+        Returns:
+            string: The script of the inventory script.
+
+        """
+        return self._data.get('script')
+
+    @script.setter
+    def script(self, value):
+        """Update the script of the inventory script.
+
+        Returns:
+            None:
+
+        """
+        self._update_values('script', value)
+
+    def create_inventory_script(self, name, description, script, organization):
+        """Creates a custom inventory script.
+
+        Args:
+            name: Name of the inventory script.
+            description: The description of the inventory script.
+            script: The script of the inventory script.
+            organization: The organization the inventory script is part of.
+
+        Returns:
+            Inventory_script: The created inventory script is successful, None otherwise.
+
+        """
+        url = '{api}/inventory_scripts/'.format(api=self._tower.api)
+        payload = {'name': name,
+                   'description': description,
+                   'inventory': self.id,
+                   'script': script,
+                   'organization': organization
+                   }
+        response = self._tower.session.post(url, json=payload)
+        if not response.ok:
+            self._logger.error('Error creating host "%s", response was "%s"', name, response.text)
+        return response.json() if response.ok else None
```

Comment:
If the recursion is intentional, shouldn't we return a proper object here and not the raw json response?